### PR TITLE
Food gets stink lines when about to go rotten

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

Title says it all, food that decomposes gets stink lines halfway through.
image
image

Moldy messes always gets stink lines.

## Why It's Good For The Game

1. It's funny
2. It's a visual indicator for food going bad which is kinda nice

## Changelog

:cl:
add: Food now gets stink lines when going bad. Uh oh, stinky.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
quite pungent

**Comment by Singul0** [comment]
brap fetish, no fetish content is allowed in this codebase, especially not brap.

**Review by Wallemations — APPROVED** [review]
As the person that made decomposition I can't believe I didn't think of this

**Comment by ArcaneMusic** [comment]
Might I suggest a different visual, like perhaps a little clock, or maybe flies instead?

**Comment by ChungusGamer666** [comment]
> Might I suggest a different visual, like perhaps a little clock, or maybe flies instead?

no, stink lines are funnier

**Comment by ChungusGamer666** [comment]
> no, stink lines are funnier

why are you booing me

**Comment by BlueMemesauce** [comment]
The issue is that it just looks like a green worm in the bread
Shouldnt the stink lines be surrounding it rather than touching it? https://www.youtube.com/watch?v=4ILKyggzDLM

**Comment by ChungusGamer666** [comment]
> The issue is that it just looks like a green worm in the bread Shouldnt the stink lines be surrounding it rather than touching it? https://www.youtube.com/watch?v=4ILKyggzDLM

it's particles, it floats up :)
i was gonna make an mp4 showing it instead of a screenshot but i am a lazy ass

File: code/datums/components/food/decomposition.dm [hunk 1]
```diff
@@ -53,8 +48,26 @@
...
+/datum/component/decomposition/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, PROC_REF(handle_movement))
+	RegisterSignals(parent, list(
+		COMSIG_ITEM_PICKUP, //person picks up an item
+		COMSIG_ATOM_ENTERED), //Object enters a storage object (boxes, etc.)
+		PROC_REF(picked_up))
+	RegisterSignals(parent, list(
+		COMSIG_ITEM_DROPPED, //Object is dropped anywhere
+		COMSIG_ATOM_EXITED), //Object exits a storage object (boxes, etc)
+		PROC_REF(dropped))
```
Comment (san7890) [inline code/datums/components/food/decomposition.dm:69]:
```suggestion
		PROC_REF(dropped),
	)
```

File: code/datums/components/food/decomposition.dm [hunk 2]
```diff
@@ -53,8 +48,26 @@
...
+	. = ..()
+	if(particle_effect)
+		QDEL_NULL(particle_effect)
+
+/datum/component/decomposition/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, PROC_REF(handle_movement))
+	RegisterSignals(parent, list(
+		COMSIG_ITEM_PICKUP, //person picks up an item
+		COMSIG_ATOM_ENTERED), //Object enters a storage object (boxes, etc.)
+		PROC_REF(picked_up))
```
Comment (san7890) [inline code/datums/components/food/decomposition.dm:65]:
```suggestion
	RegisterSignals(parent, list(
		COMSIG_ITEM_PICKUP, //person picks up an item
		COMSIG_ATOM_ENTERED), //Object enters a storage object (boxes, etc.)
		PROC_REF(picked_up),
	)
```

multiline should be odne correctly

**Review by san7890 — CHANGES_REQUESTED** [review]

**Review by Wallemations — APPROVED** [review]
much better

**Pull request merged**